website/api/controllers/try-fleet/view-query-report.js

```diff
@@ -0,0 +1,237 @@
+      isIn: ['macos', 'linux', 'windows']
+    },
+
+    tableName: {
+      type: 'string',
+      required: true,
+      description: 'The name of the osquery table to show results for.',
+    },
+
+  },
+
+
+  exits: {
+
+    success: {
+      viewTemplatePath: 'pages/try-fleet/query-report'
+    },
+
+    badConfig: {
+      responseType: 'badConfig'
+    },
+
+    redirect: {
+      description: 'The requesting user is not logged in.',
+      responseType: 'redirect'
+    },
+
+    invalidTable: {
+      responseType: 'notFound',
+      description: 'No osquery table with the specified name could be found.'
+    },
+
+  },
+
+
+  fn: async function ({hostPlatform, tableName}) {
+
+    if(!_.isObject(sails.config.builtStaticContent) || !_.isArray(sails.config.builtStaticContent.osqueryTables)){
+      throw {badConfig: 'builtStaticContent.osqueryTables'};
+    }
+
+    // If the requesting user is not logged in, redirect them to the /try-fleet/login page with the specified hostPlatform added as a query parameter.
+    if(!this.req.me){
+      throw {redirect: `/try-fleet/login?targetPlatform=${hostPlatform}` };
+    }
+
+    if(!sails.config.custom.queryIdsByTableName){
+      throw new Error('Missing config variable: The dictionary of query ids required to use the query-report page is missing! (sails.config.custom.queryIdsByTableName)');
+    }
+
+    if(!sails.config.custom.hostIdsByHostPlatform){
+      throw new Error('Missing config variable: The dictionary of host ids required to use the query-report page is missing! (sails.config.custom.hostIdsByHostPlatform)');
+    }
+
+    if(!sails.config.custom.teamApidForQueryReports){
+      throw new Error('Missing config variable: The id of the team the query report page gets results for is missing! (sails.config.custom.teamApidForQueryReports)');
+    }
+
+    if(!sails.config.custom.fleetBaseUrlForQueryReports){
+      throw new Error('Missing config variable: The URL of the fleet instance used for query reports is missing! (sails.config.custom.fleetBaseUrlForQueryReports)');
+    }
+
+    if(!sails.config.custom.fleetTokenForQueryReports){
+      throw new Error('Missing config variable: The API token for requests to the Fleet instance used for queyr reports is missing! (sails.config.custom.fleetTokenForQueryReports)');
+    }
+
+    //  ┬ ┬┌─┐┌─┐┌┬┐  ┬┌┐┌┌─┐┌─┐┬─┐┌┬┐┌─┐┌┬┐┬┌─┐┌┐┌
+    //  ├─┤│ │└─┐ │   ││││├┤ │ │├┬┘│││├─┤ │ ││ ││││
+    //  ┴ ┴└─┘└─┘ ┴   ┴┘└┘└  └─┘┴└─┴ ┴┴ ┴ ┴ ┴└─┘┘└┘
+    let hostIdsByHostPlatform = sails.config.custom.hostIdsByHostPlatform;
+    // Get the ID of the host we'll be showing results for.
+    let selectedHostId = hostIdsByHostPlatform[hostPlatform];
+
+    // Send an HTTP request to get the host details for hosts on the query report team.
+    let hostsOnQueryReportTeamApiResponse = await sails.helpers.http.get.with({
+      url: sails.config.custom.fleetBaseUrlForQueryReports+'/api/v1/fleet/hosts?team_id='+sails.config.custom.teamApidForQueryReports,
```

Comment:
even though it's config, do encodeURICompoennt for the part after the querystring

(not for the baseurl)